File: shoring/flywheel/src/test/java/com/panopset/tests/flywheel/SplitColumnListTest.java
package com.panopset.tests.flywheel;

import static org.junit.jupiter.api.Assertions.assertEquals;
import java.io.IOException;
import org.junit.jupiter.api.Test;
import com.panopset.compat.Stringop;
import com.panopset.flywheel.FlywheelBuilder;
import com.panopset.flywheel.LineFeedRules;
import com.panopset.flywheel.ReservedWords;

public final class SplitColumnListTest {

  /**
   * Test split columns in a list command. See <b>splitList.txt</b>.
   */
  @Test
  void testSplitColumns() throws IOException {
    Stringop.setEol("\n");
    String expected = "6#Zonk7#Bonk\n";
    String[] inp = new String[] { "${@l splitList.txt}${2}#${3}${@q}" };
    String results = new FlywheelBuilder().withLineFeedRules(LineFeedRules.LINE_BREAKS)
    		// TODO use testing framework.
        .baseDirectoryPath("src/test/resources/com/panopset/tests/flywheel")
        .map(ReservedWords.SPLITS, "13,14").input(inp).construct().exec();
    assertEquals(expected, results);
  }

}

File: code/shoring/flywheel/src/test/kotlin/com/panopset/tests/flywheel/SplitColumnListTest.kt
package com.panopset.tests.flywheel

import com.panopset.compat.Stringop.setEol
import com.panopset.flywheel.FlywheelBuilder
import com.panopset.flywheel.LINE_BREAKS
import com.panopset.flywheel.ReservedWords
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

class SplitColumnListTest {
    /**
     * Test split columns in a list command. See **splitList.txt**.
     */
    @Test
    fun testSplitColumns() {
        setEol("\n")
        val expected = "6#Zonk7#Bonk"
        val inp = arrayOf<String?>("\${@l splitList.txt}\${2}#\${3}\${@q}")
        val results = FlywheelBuilder().withLineFeedRules(LINE_BREAKS) // TODO use testing framework.
            .baseDirectoryPath("src/test/resources/com/panopset/tests/flywheel")
            .map(ReservedWords.SPLITS, "13,14").inputArray(inp).construct().exec()
        Assertions.assertEquals(expected, results)
    }
}
